### PR TITLE
HACDOCS-435: Creating README content for Go code sample.

### PR DESCRIPTION
**Pull request by jseseCCS**
[HACDOCS-435](https://issues.redhat.com/browse/HACDOCS-435) creating README content for Go code sample.

### CODE REVIEW

**Comment by jseseCCS** [comment]
@maysunfaisal this one is ready for your input, please

File: README.md [hunk 1]
```diff
@@ -1 +1,16 @@
-# devfile-sample-go-basic
\ No newline at end of file
+# sample-devfile-with-go-basic
+
+The Go sample uses the **HTTP 8081** port. 
+
+## Defining the workspace lifecycle with a devfile.yaml file
+
+Use the `devfile.yaml` file to
```
Comment (maysunfaisal) [inline README.md:7]:
```suggestion
Use the `devfile.yaml` file to  define the workspace for you application developement lifecycle
```

**Comment by maysunfaisal** [comment]
I think generally we can follow the same README pattern as what we are doing in https://github.com/devfile-samples/devfile-sample-java-springboot-basic/pull/25 except it should be catered to Go where ever necessary

**Comment by jseseCCS** [comment]
@maysunfaisal can you review this one? thank you!

File: README.md [hunk 2]
```diff
@@ -1 +1,19 @@
...
+**Note:** The Go code sample uses the **8081** HTTP port.
+
+Before you begin creating an application with this `devfile` code sample, it's helpful to understand the relationship between the `devfile` and `Dockerfile` and how they contribute to your build. You can find these files at the following URLs:
+
+* [Go `devfile.yaml`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/devfile.yaml)
+* [Go `Dockerfile`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/docker/Dockerfile)
+
+1. The `devfile.yaml` file has an [`image-build` component](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/devfile.yaml#L21-L27) that points to your `Dockerfile`.
+2. The [`docker/Dockerfile`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/docker/Dockerfile) contains the instructions you need to build the code sample as a container image.
+3. The `devfile.yaml` [`kubernetes-deploy` component](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/devfile.yaml#L28-L41) points to a `deploy.yaml` file that contains instructions for deploying the built container image.
```
Comment (maysunfaisal) [inline README.md:12]:
```suggestion
3. The `devfile.yaml` [`kubernetes-deploy` component](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/devfile.yaml#L27-L40) points to a `deploy.yaml` file that contains instructions for deploying the built container image.
```

File: README.md [hunk 3]
```diff
@@ -1 +1,19 @@
...
+# Creating an application with a Go code sample
+
+**Note:** The Go code sample uses the **8081** HTTP port.
+
+Before you begin creating an application with this `devfile` code sample, it's helpful to understand the relationship between the `devfile` and `Dockerfile` and how they contribute to your build. You can find these files at the following URLs:
+
+* [Go `devfile.yaml`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/devfile.yaml)
+* [Go `Dockerfile`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/docker/Dockerfile)
+
+1. The `devfile.yaml` file has an [`image-build` component](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/devfile.yaml#L21-L27) that points to your `Dockerfile`.
```
[the remark this answers is not in the document]
Comment (jseseCCS) [reply README.md]: yes! these are the very changes i wanted to make before merge. done and ty! did other similar ones too

File: README.md [hunk 4]
```diff
@@ -1 +1,19 @@
...
+2. The [`docker/Dockerfile`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/docker/Dockerfile) contains the instructions you need to build the code sample as a container image.
+3. The `devfile.yaml` [`kubernetes-deploy` component](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/devfile.yaml#L28-L41) points to a `deploy.yaml` file that contains instructions for deploying the built container image.
+4. The `devfile.yaml` [`deploy` command](https://github.com/devfile-samples/devfile-sample-go-basic/blob/main/devfile.yaml#L43-L56) completes the [outerloop](https://devfile.io/docs/2.2.0/innerloop-vs-outerloop) deployment phase by pointing to the `image-build` and `kubernetes-deploy` components to create your application.
```
[the remark this answers is not in the document]
Comment (jseseCCS) [reply README.md]: what about 44-55?

[the remark this answers is not in the document]
Comment (maysunfaisal) [reply README.md]: it starts at line 42

**Pull request merged**